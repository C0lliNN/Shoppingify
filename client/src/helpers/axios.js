import axios from 'axios';
import store from '../store/index';

export default function getAxios() {
  const config = {
    headers: {},
  };
  // eslint-disable-next-line no-undef
  if (process.env.NODE_ENV === 'development') {
    config.baseURL = 'http://localhost:5000/api/v1';
  } else {
    config.baseURL = '/api/v1';
  }

  config.headers['Content-Type'] = 'application/json';

  const token = store.getState().auth.token;

  if (token) {
    config.headers['Authorization'] = `Bearer ${token}`;
  }

  return axios.create(config);
}
