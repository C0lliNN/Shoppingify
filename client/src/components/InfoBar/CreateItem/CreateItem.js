import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import * as variables from '../../../helpers/style-constants';
import ButtonBar from '../../ButtonBar/ButtonBar';
import Button from '../../UI/Button/Button';
import FromGroup from '../../UI/FormGroup/FormGroup';
import CategoryFormGroup from '../../CategoryFormGroup/CategoryFormGroup';
import { useState } from 'react';
import { showListBuilder } from '../../../store/actions';
import { connect } from 'react-redux';
import Modal from '../../UI/Modal/Modal';
import getAxios from '../../../helpers/axios';

const StyledCreateItem = styled.div`
  padding: 20px 15px;
`;

const Title = styled.h3`
  font-size: ${variables.FONT_SIZE_4};
  font-weight: 600;
  margin-bottom: 30px;
`;

function validateInput(data) {
  const { name, note, image, category } = data;
  if (!name) {
    return 'The name is required';
  }
  if (name.length < 2 || name.length > 40) {
    return 'The name must contain between 2 and 40 chars';
  }
  if (note && note.length > 255) {
    return 'The note must contain at most 255 chars';
  }
  //prettier-ignore
  //eslint-disable-next-line
  if (image && !image.match(/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i)){
    return "The image must be a valid url"
  }
  if (!category) {
    return 'The category is required';
  }
  if (category.length < 4 || category.length > 120) {
    return 'The category must contain between 4 and 120 chars';
  }

  return null;
}

function CreateItem({ categories, showListBuilder }) {
  const [name, setName] = useState('');
  const [note, setNote] = useState('');
  const [image, setImage] = useState('');
  const [category, setCategory] = useState('');

  const [showModal, setShowModal] = useState(false);
  const [modalTitle, setModalTitle] = useState(null);

  async function handleSubmit(event) {
    event.preventDefault();

    const payload = { name, note, image, category };

    const error = validateInput(payload);

    if (error) {
      setModalTitle(error);
      setShowModal(true);
    } else {
      try {
        const response = await getAxios().post('/itens', payload);
        console.log(response.data);
      } catch (error) {
        console.error(error);
      }
    }
  }

  function closeModal() {
    setShowModal(false);
  }

  return (
    <StyledCreateItem>
      <form onSubmit={handleSubmit} action="#" method="post">
        <Title>Add a new item</Title>
        <FromGroup>
          <FromGroup.Label>Name</FromGroup.Label>
          <FromGroup.Input
            placeholder="Enter a name"
            name="name"
            value={name}
            onChange={(event) => setName(event.target.value)}
          />
        </FromGroup>
        <FromGroup>
          <FromGroup.Label>Note(optional)</FromGroup.Label>
          <FromGroup.Textarea
            placeholder="Enter a note"
            rows="3"
            name="note"
            value={note}
            onChange={(event) => setNote(event.target.value)}
          ></FromGroup.Textarea>
        </FromGroup>
        <FromGroup>
          <FromGroup.Label>Image(optional)</FromGroup.Label>
          <FromGroup.Input
            placeholder="Enter a url"
            name="image"
            value={image}
            onChange={(event) => setImage(event.target.value)}
          />
        </FromGroup>
        <CategoryFormGroup
          categories={categories}
          categoryValue={category}
          setCategoryValue={setCategory}
        />
        <ButtonBar>
          <Button btnType="flat" onClick={showListBuilder}>
            cancel
          </Button>
          <Button btnType="raised" variant="primary" type="submit">
            Save
          </Button>
        </ButtonBar>
      </form>
      {showModal && (
        <Modal
          title={modalTitle}
          onClose={closeModal}
          okButton={
            <Button btnType="raised" variant="danger" onClick={closeModal}>
              OK
            </Button>
          }
        />
      )}
    </StyledCreateItem>
  );
}

CreateItem.propTypes = {
  categories: PropTypes.array,
  showListBuilder: PropTypes.func,
};

const mapDispatchToProps = {
  showListBuilder,
};

const mapStateToProps = (state) => {
  return {
    categories: state.activeList.itemsGroup.map((group) => group.category),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateItem);
