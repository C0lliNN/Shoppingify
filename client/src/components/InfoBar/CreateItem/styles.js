import styled from 'styled-components';
import * as variables from '../../../helpers/style-constants';

export const StyledCreateItem = styled.div`
  padding: 20px 15px;
  box-sizing: border-box;
  position: relative;
  overflow-y: auto;
  &::-webkit-scrollbar {
    display: none;
  }
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    height: 100%;
  }
`;

export const Title = styled.h3`
  font-size: ${variables.FONT_SIZE_4};
  font-weight: 600;
  margin-bottom: 30px;
`;

export const ValidationError = styled.span`
  font-size: 12px;
  font-weight: 600;
  color: ${variables.COLORS.danger};
  align-self: start;
  display: block;
  margin-top: 8px;
`;
