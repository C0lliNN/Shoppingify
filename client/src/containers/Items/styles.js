import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

export const StyledItems = styled.div`
  padding: 20px;
  box-sizing: border-box;
  min-height: 100%;
  background-color: #fafafe;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    padding: 20px 60px;
  }
`;

export const Header = styled.div`
  display: flex;
  align-items: top;
  justify-content: start;
  flex-wrap: wrap;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    flex-wrap: nowrap;
    justify-content: space-around;
  }
`;

export const Text = styled.p`
  font-size: ${variables.FONT_SIZE_3};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  margin-top: 50px;
`;