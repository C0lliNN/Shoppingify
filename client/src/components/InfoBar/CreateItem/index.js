import React from 'react';
import ButtonBar from '../../ButtonBar';
import Button from '../../UI/Button';
import FormGroup from '../../UI/FormGroup';
import CategoryFormGroup from '../../CategoryFormGroup';
import { useState } from 'react';
import { showListBuilder } from '../../../store/actions';
import { useDispatch, useSelector } from 'react-redux';
import Modal from '../../UI/Modal';
import getAxios from '../../../helpers/axios';
import Spinner from '../../UI/Spinner/Spinner';
import { addItem } from '../../../store/actions';
import { StyledCreateItem, Title, ValidationError } from './styles';
import { useForm } from 'react-hook-form';

function CreateItem() {
  const [isLoading, setIsLoading] = useState(false);

  const [showModal, setShowModal] = useState(false);
  const [modalTitle, setModalTitle] = useState(null);

  const dispatch = useDispatch();
  const categories = useSelector((state) =>
    state.itemsData.data.map((group) => group.category)
  );
  const { register, handleSubmit, errors } = useForm();

  const nameRef = register({
    required: { value: true, message: 'The name is required' },
    minLength: {
      value: 2,
      message: 'The name must contain between 2 and 40 chars',
    },
    maxLength: {
      value: 40,
      message: 'The name must contain between 2 and 40 chars',
    },
  });

  const noteRef = register({
    maxLength: {
      value: 255,
      message: 'The note must contain at most 255 chars',
    },
  });

  const imageRef = register({
    pattern: {
      value: /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i,
      message: 'The image must be a valid url',
    },
  });

  const categoryRef = register({
    required: { value: true, message: 'The category is required' },
    minLength: {
      value: 4,
      message: 'The category must contain between 4 and 120 chars',
    },
    maxLength: {
      value: 120,
      message: 'The category must contain between 4 and 120 chars',
    },
  });

  async function onSubmit(data, event) {
    event.preventDefault();

    const { name, note, category, image } = data;

    let categoryPayload = categories.find(
      (categoryItem) => categoryItem.name === category
    );

    if (!categoryPayload) {
      categoryPayload = { name: category };
    }

    const payload = {
      name,
      note,
      image,
      category: categoryPayload,
    };

    try {
      const response = await getAxios().post('/items', payload);
      dispatch(addItem(response.data));

      dispatch(showListBuilder());
    } catch (error) {
      if (error.response) {
        setModalTitle(error.response.data.message);
      } else {
        setModalTitle(error.message);
      }
      setShowModal(true);
    } finally {
      setIsLoading(false);
    }
  }

  function closeModal() {
    setShowModal(false);
  }

  console.log(errors);

  return (
    <StyledCreateItem>
      <form onSubmit={handleSubmit(onSubmit)} action="#" method="post">
        <Title>Add a new item</Title>
        {isLoading ? (
          <Spinner />
        ) : (
          <React.Fragment>
            <FormGroup>
              <FormGroup.Label htmlFor="name">Name</FormGroup.Label>
              <FormGroup.Input
                placeholder="Enter a name"
                ref={nameRef}
                id="name"
                name="name"
              />
              {errors?.name && (
                <ValidationError>{errors?.name.message}</ValidationError>
              )}
            </FormGroup>

            <FormGroup>
              <FormGroup.Label htmlFor="note">Note(optional)</FormGroup.Label>
              <FormGroup.Textarea
                placeholder="Enter a note"
                id="note"
                rows="3"
                name="note"
                ref={noteRef}
              ></FormGroup.Textarea>
              {errors?.note && (
                <ValidationError>{errors?.note.message}</ValidationError>
              )}
            </FormGroup>
            <FormGroup>
              <FormGroup.Label htmlFor="image">Image(optional)</FormGroup.Label>
              <FormGroup.Input
                placeholder="Enter a url"
                name="image"
                id="image"
                ref={imageRef}
              />
              {errors?.image && (
                <ValidationError>{errors?.image.message}</ValidationError>
              )}
            </FormGroup>
            <CategoryFormGroup
              categories={categories}
              ref={categoryRef}
              errors={errors}
            />
          </React.Fragment>
        )}
        <ButtonBar>
          <Button
            btnType="flat"
            onClick={() => dispatch(showListBuilder())}
            type="button"
          >
            cancel
          </Button>
          <Button btnType="raised" variant="primary" type="submit">
            Save
          </Button>
        </ButtonBar>
      </form>
      {showModal && (
        <Modal
          title={modalTitle}
          onClose={closeModal}
          okButton={
            <Button btnType="raised" variant="danger" onClick={closeModal}>
              OK
            </Button>
          }
        />
      )}
    </StyledCreateItem>
  );
}

export default CreateItem;
