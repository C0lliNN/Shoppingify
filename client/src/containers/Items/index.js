import React from 'react';
import Title from '../../components/Title';
import SearchBar from '../../components/UI/SearchBar';
import ItemsGroup from '../../components/ItemsGroup';
import { useSelector } from 'react-redux';
import Spinner from '../../components/UI/Spinner';
import ErrorMessage from '../../components/ErrorMessage';
import { Text, StyledItems, Header } from './styles';

function itemsReducer(items, group) {
  return items.concat(group.items.map((item) => item._id));
}

function Items() {
  const { data, isLoading, error } = useSelector((state) => state.itemsData);
  const itemsAdded = useSelector((state) =>
    state.activeList.data.reduce(itemsReducer, [])
  );

  let content = null;

  if (isLoading) {
    content = <Spinner />;
  } else if (error) {
    content = <ErrorMessage message={error} />;
  } else if (data.length) {
    content = data
      .filter((group) => group.category._id)
      .map((group) => (
        <ItemsGroup
          key={group.category._id}
          category={group.category}
          items={group.items.filter((item) => !itemsAdded.includes(item._id))}
        />
      ));
  } else {
    content = <Text>No items yet</Text>;
  }

  return (
    <StyledItems>
      <Header>
        <Title />
        <SearchBar />
      </Header>
      {content}
    </StyledItems>
  );
}

export default Items;
