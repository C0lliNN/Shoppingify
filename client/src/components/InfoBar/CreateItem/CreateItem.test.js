import React from 'react';
import 'jest-styled-components';
import { render } from '../../../tests/utilities';
import Items from '../../../containers/Items/Items';
import CreateItem from './CreateItem';
import axios from 'axios';
import { fireEvent, waitForElement } from '@testing-library/react';
import { lorem } from 'faker';

const payload = {
  name: null,
  note: null,
  image: null,
  category: null,
};

function exec() {
  const { getByLabelText, getByText, container } = render(
    <React.Fragment>
      <Items />
      <CreateItem />
    </React.Fragment>
  );

  fireEvent.change(getByLabelText('Name'), { target: { value: payload.name } });
  fireEvent.change(getByLabelText('Note(optional)'), {
    target: { value: payload.note },
  });
  fireEvent.change(getByLabelText('Image(optional)'), {
    target: { value: payload.image },
  });
  fireEvent.change(getByLabelText('Category'), {
    target: { value: payload.category },
  });
  fireEvent.click(getByText('Save'));

  return { getByLabelText, getByText, container };
}

jest.mock('axios');

describe('<CreateItem/>', () => {
  beforeEach(() => {
    payload.name = 'Banana';
    payload.note = 'The Best Banana';
    payload.image = 'https://i.imgur.com/9o7jc9o.jpeg';
    payload.category = 'Fruit';

    axios.create.mockImplementationOnce(() => {
      return {
        post: () =>
          Promise.resolve({
            data: {
              _id: '5f5543b2267b3e04c0a91682',
              name: 'Banana',
              note: 'The Best Banana',
              image: 'https://i.imgur.com/9o7jc9o.jpeg',
              category: { _id: '5f5543ab267b3e04c0a9167f', name: 'Fruit' },
              user: '5f528e6790213f0b78431156',
              __v: 0,
            },
          }),
        get: () =>
          Promise.resolve({
            data: [
              {
                _id: '5f5543b2267b3e04c0a91680',
                name: 'Apple',
                note: '',
                image: '',
                category: { _id: '5f5543ab267b3e04c0a9167f', name: 'Fruit' },
                user: '5f528e6790213f0b78431156',
                __v: 0,
              },
            ],
          }),
      };
    });
  });
  it('should render correctly', () => {
    const { container } = render(<CreateItem />);
    expect(container).toMatchSnapshot();
  });
  it('should show an error modal if name is empty', () => {
    payload.name = '';

    const { getByText } = exec();

    expect(getByText('The name is required')).not.toBeNull();
  });
  it('should show an error modal if the name.length is not between 2 and 40', () => {
    payload.name = 'R';

    const { getByText } = exec();

    expect(
      getByText('The name must contain between 2 and 40 chars')
    ).not.toBeNull();
  });
  it('should show an error modal if the note.length > 255', () => {
    payload.note = lorem.sentence(150);

    const { getByText } = exec();

    expect(getByText('The note must contain at most 255 chars')).not.toBeNull();
  });
  it('should show an error modal if image is not a valid url', () => {
    payload.image = '11111111';

    const { getByText } = exec();

    expect(getByText('The image must be a valid url')).not.toBeNull();
  });
  it('should shown an error modal if category is empty', () => {
    payload.category = '';

    const { getByText } = exec();

    expect(getByText('The category is required')).not.toBeNull();
  });
  it('should show an error modal if the category.length is not between 4 and 120', () => {
    payload.category = 'Ra';

    const { getByText } = exec();

    expect(
      getByText('The category must contain between 4 and 120 chars')
    ).not.toBeNull();
  });
  it('should add the item correctly if the payload is valid', async () => {
    const { getByText, container } = exec();

    await waitForElement(() => getByText(payload.name));
    expect(container).toMatchSnapshot();
  });
});
