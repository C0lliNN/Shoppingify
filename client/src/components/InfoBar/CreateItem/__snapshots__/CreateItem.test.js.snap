// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateItem/> should add the item correctly if the payload is valid 1`] = `
.c2 {
  font-family: Quicksand;
  font-size: 18px;
  font-weight: 600;
}

.c4 {
  background-color: #FFFFFF;
  box-shadow: 0px 2px 12px rgba(0,0,0,0.04);
  border-top-left-radius: 12px;
  border-bottom-left-radius: 12px;
  height: 100%;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 10px;
}

.c5 {
  color: #34333A;
  font-size: 24px;
}

.c6 {
  background-color: #FFFFFF;
  box-shadow: 0px 2px 12px rgba(0,0,0,0.04);
  border-top-right-radius: 12px;
  border-bottom-right-radius: 12px;
  font-family: Quicksand;
  font-weight: 600;
  height: 100%;
  border: none;
  outline: none;
}

.c6::-webkit-input-placeholder {
  color: #BDBDBD;
}

.c6::-moz-placeholder {
  color: #BDBDBD;
}

.c6:-ms-input-placeholder {
  color: #BDBDBD;
}

.c6::placeholder {
  color: #BDBDBD;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 40px;
  margin-top: 10px;
}

.c7 {
  font-family: Quicksand;
  font-size: 18px;
  color: #000000;
  margin-top: 0px;
  margin-bottom: 15px;
  font-weight: 600;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c9 {
  background-color: #FFFFFF;
  border-radius: 12px;
  font-size: 14px;
  font-weight: 600;
  box-shadow: 0px 2px 12px rgba(0,0,0,0.05);
  padding: 8px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 80px;
  margin-bottom: 15px;
  margin-right: 5px;
}

.c10 {
  cursor: pointer;
}

.c11 {
  font-size: 18px;
  color: #c1c1c4;
  cursor: pointer;
}

.c0 {
  padding: 20px;
  box-sizing: border-box;
  min-height: 100%;
  background-color: #fafafe;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c20 {
  padding: 20px 0px;
  display: block;
  background: transparent;
  text-align: center;
}

.c21 {
  background-color: transparent;
  border-radius: 12px;
  color: #34333A;
  border-color: transparent;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c21:disabled {
  color: #BDBDBD   border-color:#BDBDBD;
  cursor: default;
}

.c22 {
  background-color: #F9A109;
  border-color: #F9A109;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c22:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 25px;
}

.c15 {
  display: block;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  width: 100%;
  box-sizing: border-box;
  padding: 20px 12px;
  border: 2px solid #BDBDBD;
  border-radius: 12px;
  outline: none;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: 500;
  color: #34333A;
}

.c15:focus {
  border-color: #F9A109;
  box-shadow: 0px 0px 1px rgba(0,0,0,0.3);
}

.c15:focus +label {
  color: #F9A109;
}

.c15::-webkit-input-placeholder {
  color: #BDBDBD;
}

.c15::-moz-placeholder {
  color: #BDBDBD;
}

.c15:-ms-input-placeholder {
  color: #BDBDBD;
}

.c15::placeholder {
  color: #BDBDBD;
}

.c17 {
  display: block;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  width: 100%;
  box-sizing: border-box;
  padding: 20px 12px;
  border: 2px solid #BDBDBD;
  border-radius: 12px;
  outline: none;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: 500;
  color: #34333A;
}

.c17:focus {
  border-color: #F9A109;
  box-shadow: 0px 0px 1px rgba(0,0,0,0.3);
}

.c17:focus +label {
  color: #F9A109;
}

.c17::-webkit-input-placeholder {
  color: #BDBDBD;
}

.c17::-moz-placeholder {
  color: #BDBDBD;
}

.c17:-ms-input-placeholder {
  color: #BDBDBD;
}

.c17::placeholder {
  color: #BDBDBD;
}

.c16 {
  display: block;
  width: 100%;
  color: #34333A;
  font-weight: 600;
  margin-bottom: 5px;
  font-size: 14px;
}

.c18 {
  border: 1px solid #E0E0E0;
  box-sizing: border-box;
  padding: 12px;
  list-style: none;
  box-shadow: 0px 2px 12px rgba(0,0,0,0.04);
  border-top-left-radius: 12px;
  border-bottom-left-radius: 12px;
  height: 145px;
  overflow-y: auto;
}

.c18::-webkit-scrollbar {
  width: 5px;
}

.c18::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-top-right-radius: 12px;
  border-bottom-right-radius: 12px;
}

.c18::-webkit-scrollbar-thumb {
  background: #E0E0E0;
  border-radius: 12px;
}

.c18::-webkit-scrollbar-thumb:hover {
  background: #BDBDBD;
}

.c19 {
  font-size: 18px;
  line-height: 22px;
  padding: 10px;
  box-sizing: border-box;
  border-radius: 12px;
  font-weight: 500;
  font-family: Quicksand;
  color: #828282;
  cursor: pointer;
}

.c19:hover,
.c19:active {
  background-color: #f2f2f2;
  color: #000000;
}

.c12 {
  padding: 20px 15px;
  box-sizing: border-box;
  position: relative;
  overflow-y: auto;
}

.c12::-webkit-scrollbar {
  display: none;
}

.c13 {
  font-size: 24px;
  font-weight: 600;
  margin-bottom: 30px;
}

@media (min-width:1200px) {
  .c2 {
    font-size: 24px;
  }
}

@media (min-width:1200px) {
  .c3 {
    font-size: 24px;
    margin-left: 20px;
    margin-top: 0px;
  }
}

@media (min-width:1200px) {
  .c9 {
    width: 120px;
    margin-right: 15px;
    margin-bottom: 30px;
    font-size: 18px;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 20px 60px;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (min-width:768px) {
  .c20 {
    position: fixed;
    bottom: 0;
    right: 0;
    background: #fefefe;
    width: 250px;
  }
}

@media (min-width:1200px) {
  .c20 {
    width: 370px;
  }
}

@media (min-width:768px) {
  .c18 {
    margin-bottom: 100px;
  }
}

@media (min-width:768px) {
  .c12 {
    height: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        <span
          style="color: rgb(249, 161, 9); font-weight: bold;"
        >
          Shoppingify
           
        </span>
        allows you take your shopping list wherever you go
      </h1>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5 material-icons-round"
          >
            search
          </div>
        </div>
        <input
          class="c6"
          placeholder="Search item"
        />
      </div>
    </div>
    <div
      style="margin-top: 50px;"
    >
      <h3
        class="c7"
      >
        Fruit
      </h3>
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <span
            class="c10"
          >
            Apple
          </span>
          <i
            class="c11 material-icons-round"
          >
            add
          </i>
        </div>
        <div
          class="c9"
        >
          <span
            class="c10"
          >
            Banana
          </span>
          <i
            class="c11 material-icons-round"
          >
            add
          </i>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c12"
  >
    <form
      action="#"
      method="post"
    >
      <h3
        class="c13"
      >
        Add a new item
      </h3>
      <div
        class="c14"
      >
        <input
          class="c15"
          id="name"
          name="name"
          placeholder="Enter a name"
          value="Banana"
        />
        <label
          class="c16"
          for="name"
        >
          Name
        </label>
      </div>
      <div
        class="c14"
      >
        <textarea
          class="c17"
          id="note"
          name="note"
          placeholder="Enter a note"
          rows="3"
        >
          The Best Banana
        </textarea>
        <label
          class="c16"
          for="note"
        >
          Note(optional)
        </label>
      </div>
      <div
        class="c14"
      >
        <input
          class="c15"
          id="image"
          name="image"
          placeholder="Enter a url"
          value="https://i.imgur.com/9o7jc9o.jpeg"
        />
        <label
          class="c16"
          for="image"
        >
          Image(optional)
        </label>
      </div>
      <div
        class="c14"
      >
        <input
          class="c15"
          id="category"
          placeholder="Enter a Category"
          value="Fruit"
        />
        <label
          class="c16"
          for="category"
        >
          Category
        </label>
      </div>
      <ul
        class="c18"
        style="visibility: hidden;"
      >
        <li
          class="c19"
        >
          Fruit
        </li>
      </ul>
      <div
        class="c20"
      >
        <button
          class="c21"
          type="button"
        >
          cancel
        </button>
        <button
          class="c22"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<CreateItem/> should render correctly 1`] = `
.c7 {
  padding: 20px 0px;
  display: block;
  background: transparent;
  text-align: center;
}

.c8 {
  background-color: transparent;
  border-radius: 12px;
  color: #34333A;
  border-color: transparent;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c8:disabled {
  color: #BDBDBD   border-color:#BDBDBD;
  cursor: default;
}

.c9 {
  background-color: #F9A109;
  border-color: #F9A109;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 20px;
  border-width: 2px;
  border-style: solid;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: bold;
  outline: none;
  cursor: pointer;
}

.c9:disabled {
  background-color: #BDBDBD;
  border-color: #BDBDBD;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 25px;
}

.c3 {
  display: block;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  width: 100%;
  box-sizing: border-box;
  padding: 20px 12px;
  border: 2px solid #BDBDBD;
  border-radius: 12px;
  outline: none;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: 500;
  color: #34333A;
}

.c3:focus {
  border-color: #F9A109;
  box-shadow: 0px 0px 1px rgba(0,0,0,0.3);
}

.c3:focus +label {
  color: #F9A109;
}

.c3::-webkit-input-placeholder {
  color: #BDBDBD;
}

.c3::-moz-placeholder {
  color: #BDBDBD;
}

.c3:-ms-input-placeholder {
  color: #BDBDBD;
}

.c3::placeholder {
  color: #BDBDBD;
}

.c5 {
  display: block;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  width: 100%;
  box-sizing: border-box;
  padding: 20px 12px;
  border: 2px solid #BDBDBD;
  border-radius: 12px;
  outline: none;
  font-size: 14px;
  font-family: Quicksand;
  font-weight: 500;
  color: #34333A;
}

.c5:focus {
  border-color: #F9A109;
  box-shadow: 0px 0px 1px rgba(0,0,0,0.3);
}

.c5:focus +label {
  color: #F9A109;
}

.c5::-webkit-input-placeholder {
  color: #BDBDBD;
}

.c5::-moz-placeholder {
  color: #BDBDBD;
}

.c5:-ms-input-placeholder {
  color: #BDBDBD;
}

.c5::placeholder {
  color: #BDBDBD;
}

.c4 {
  display: block;
  width: 100%;
  color: #34333A;
  font-weight: 600;
  margin-bottom: 5px;
  font-size: 14px;
}

.c6 {
  border: 1px solid #E0E0E0;
  box-sizing: border-box;
  padding: 12px;
  list-style: none;
  box-shadow: 0px 2px 12px rgba(0,0,0,0.04);
  border-top-left-radius: 12px;
  border-bottom-left-radius: 12px;
  height: 145px;
  overflow-y: auto;
}

.c6::-webkit-scrollbar {
  width: 5px;
}

.c6::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-top-right-radius: 12px;
  border-bottom-right-radius: 12px;
}

.c6::-webkit-scrollbar-thumb {
  background: #E0E0E0;
  border-radius: 12px;
}

.c6::-webkit-scrollbar-thumb:hover {
  background: #BDBDBD;
}

.c0 {
  padding: 20px 15px;
  box-sizing: border-box;
  position: relative;
  overflow-y: auto;
}

.c0::-webkit-scrollbar {
  display: none;
}

.c1 {
  font-size: 24px;
  font-weight: 600;
  margin-bottom: 30px;
}

@media (min-width:768px) {
  .c7 {
    position: fixed;
    bottom: 0;
    right: 0;
    background: #fefefe;
    width: 250px;
  }
}

@media (min-width:1200px) {
  .c7 {
    width: 370px;
  }
}

@media (min-width:768px) {
  .c6 {
    margin-bottom: 100px;
  }
}

@media (min-width:768px) {
  .c0 {
    height: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <form
      action="#"
      method="post"
    >
      <h3
        class="c1"
      >
        Add a new item
      </h3>
      <div
        class="c2"
      >
        <input
          class="c3"
          id="name"
          name="name"
          placeholder="Enter a name"
          value=""
        />
        <label
          class="c4"
          for="name"
        >
          Name
        </label>
      </div>
      <div
        class="c2"
      >
        <textarea
          class="c5"
          id="note"
          name="note"
          placeholder="Enter a note"
          rows="3"
        />
        <label
          class="c4"
          for="note"
        >
          Note(optional)
        </label>
      </div>
      <div
        class="c2"
      >
        <input
          class="c3"
          id="image"
          name="image"
          placeholder="Enter a url"
          value=""
        />
        <label
          class="c4"
          for="image"
        >
          Image(optional)
        </label>
      </div>
      <div
        class="c2"
      >
        <input
          class="c3"
          id="category"
          placeholder="Enter a Category"
          value=""
        />
        <label
          class="c4"
          for="category"
        >
          Category
        </label>
      </div>
      <ul
        class="c6"
        style="visibility: hidden;"
      />
      <div
        class="c7"
      >
        <button
          class="c8"
          type="button"
        >
          cancel
        </button>
        <button
          class="c9"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
